master/src/main/scala/org/apache/celeborn/service/deploy/master/network/CelebornRackResolver.scala

```diff
@@ -17,33 +17,87 @@
 
 package org.apache.celeborn.service.deploy.master.network
 
+import java.io.File
+import java.util.concurrent.{ScheduledExecutorService, TimeUnit}
+
 import scala.collection.JavaConverters._
 import scala.collection.mutable.ArrayBuffer
 
 import com.google.common.base.Strings
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.CommonConfigurationKeysPublic
-import org.apache.hadoop.net.{CachedDNSToSwitchMapping, DNSToSwitchMapping, NetworkTopology, Node, NodeBase, ScriptBasedMapping}
+import org.apache.hadoop.net.{CachedDNSToSwitchMapping, DNSToSwitchMapping, NetworkTopology, Node, NodeBase, ScriptBasedMapping, TableMapping}
 import org.apache.hadoop.util.ReflectionUtils
 
 import org.apache.celeborn.common.CelebornConf
 import org.apache.celeborn.common.internal.Logging
-import org.apache.celeborn.common.util.CelebornHadoopUtils
+import org.apache.celeborn.common.util.{CelebornHadoopUtils, ThreadUtils}
 
 class CelebornRackResolver(celebornConf: CelebornConf) extends Logging {
 
+  private var rackResolveRefreshThreadPool: ScheduledExecutorService = _
+
+  private var rackResolveLastModifiedTime = 0L
+
   private val dnsToSwitchMapping: DNSToSwitchMapping = {
     val conf: Configuration = CelebornHadoopUtils.newConfiguration(celebornConf)
     val dnsToSwitchMappingClass =
       conf.getClass(
         CommonConfigurationKeysPublic.NET_TOPOLOGY_NODE_SWITCH_MAPPING_IMPL_KEY,
         classOf[ScriptBasedMapping],
         classOf[DNSToSwitchMapping])
-    ReflectionUtils.newInstance(dnsToSwitchMappingClass, conf)
-      .asInstanceOf[DNSToSwitchMapping] match {
+    val switchMapping = ReflectionUtils.newInstance(dnsToSwitchMappingClass, conf)
+      .asInstanceOf[DNSToSwitchMapping]
+    val mapping = switchMapping match {
       case c: CachedDNSToSwitchMapping => c
       case o => new CachedDNSToSwitchMapping(o)
     }
+    val refreshInterval = celebornConf.rackResolveRefreshInterval
+    rackResolveRefreshThreadPool =
+      ThreadUtils.newDaemonSingleThreadScheduledExecutor("rack-check-thread")
+    val fileName = switchMapping match {
+      case _: ScriptBasedMapping =>
+        conf.get(CommonConfigurationKeysPublic.NET_TOPOLOGY_SCRIPT_FILE_NAME_KEY)
+      case _: TableMapping =>
+        conf.get(CommonConfigurationKeysPublic.NET_TOPOLOGY_TABLE_MAPPING_FILE_KEY)
+      case _ => null
+    }
+    if (fileName != null) {
+      var scriptFile = new File(fileName)
```

Comment:
Need check file exist? or `canRead` already contain this ?

Reply:
I have made some test and canRead already contain this.